### PR TITLE
course search mobile view

### LINKED ISSUE #2103: Mobile layout for course search
As a user, I'd like to be able to search for courses on my phone. 

#### Designs and Mockups

tbd

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?

closes  #2103

#### What's this PR do?

this PR implements mobile view for the course search and course home page. in particular, there are two main things I'm doing here:

- fixing the facets, so that only mobile you have to click a little button to open them (they take over the screen when they're open)
- fix the display of the search bar and the top banner on mobile

#### How should this be manually tested?

test it out at a bunch of screen widths, both on the course home page and on the course search page, to make sure things look ok and that they are working well. I think it should all be good as far as the responsive stuff.

#### Screenshots (if appropriate)

hompe
searchfilters
search

### CODE REVIEW

**Comment by Ferdi** [comment]
Not clear from the screenshots what the user flow is , do we have an "Apply filter" button ?
https://projects.invisionapp.com/d/main#/console/18274223/379458320/preview

**Comment by aliceriot** [comment]
ah, forgot to add that, haha

**Comment by aliceriot** [comment]
sure - I hadn't bothered to change that since there's no cursor on mobile but I'll change it just incase someone has a tiny laptop or something

**Comment by aliceriot** [comment]
@mbertrand ok that should be good now

**Comment by aliceriot** [comment]
@mbertrand I forgot to add the "apply filters" button that Ferdi mentioned above, so I just put that in (and also renamed that component, which I also meant to do!)

mind taking another look?

**Comment by mbertrand** [comment]
looking at it now

**Comment by aliceriot** [comment]
yeah we have it set up to apply filters automatically, I'm just implementing what's in the mockups but maybe we want to change the behavior. @abdulkdawson what do you think?

**Comment by aliceriot** [comment]
list of changes to make from talking with @abdulkdawson yesterday (please add if I forgot any!)

- [ ] make whole filter thingy clickable
- [ ] remove image display from "list" version of card

were there some others?

**Comment by aliceriot** [comment]
ok so here's a version with the  image deleted on the list view:

removingimage


one thing I'm wondering is about widths like this one:

pictureornot

this is a "tablet width" where we switch from the one-column to the two-column layout. should we keep hiding the image on tablets too?

**Comment by aliceriot** [comment]
ok I added a final tweak, which is to change it so that at tablet widths (between 840 and 1100px) we do columns, and only go up to three columns above that width, for the grid view. this keeps the grid view cards from ever getting too squished.

@mbertrand I think this is ready for review again

**Comment by mbertrand** [comment]
@aliceriot looks good, but is it ok that the 'Apply Filters' button really just closes the filter view?

**Comment by mbertrand** [comment]
maybe "View Results" would be better for the button text?

**Comment by Ferdi** [comment]
Are the screenshot up to date ?   I still don't see the 'apply filter' button . Can you please explain a bit how is this suppose to work ?

**Comment by aliceriot** [comment]
this is what the drawer for the filters currently looks like:

applyfiltersasdf

right now the button simply closes the drawer, because our current setup is that whenever the user changes their filter selection the search is automatically re-run

**Comment by aliceriot** [comment]
I put the button like that because it's in the designs, but we may want to revisit it because it could be a little confusing for the user. i.e. if they open the drawer, change a filter, and then click the 'x' they may have the expectation that the changes they've just made haven't been applied yet, when in fact they were applied right away.

**Comment by aliceriot** [comment]
@ferdi said we can go with this for now, and I'll circle back to fix and / or change the filter behavior for mobile

**Comment by abdulkdawson** [comment]
@aliceriot how about Apply Selection(s) as the button title?

**Comment by aliceriot** [comment]
ok, two little text tweaks!

'Filter' -> 'Filter Results'
filterresults

'Apply Filters' -> 'Show Filtered Results'
filerdraweropen

how's that look?

File: static/js/components/CourseFilterDrawer_test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      const wrapper = render()
+      assert.ok(wrapper.find(".controls").exists())
+      assert.ok(wrapper.find(".open-drawer").exists())
+      assert.isNotOk(wrapper.find(".child-child-child").exists())
+    })
+  })
+
+  //
+  ;[TABLET, PHONE].forEach(deviceCategory => {
+    it(`should open the drawer whne you click the control when ${deviceCategory}`, () => {
```
Comment (mbertrand) [inline static/js/components/CourseFilterDrawer_test.js]: typo: whne _> when

**Comment by mbertrand** [comment]
@aliceriot LGTM! Some tests need fixing though

File: static/js/components/CourseFilterDrawer_test.js (hunk 1 above)
Comment (aliceriot) [reply static/js/components/CourseFilterDrawer_test.js]: lol

**Comment by aliceriot** [comment]
oopsie, fixed that stuff @mbertrand

**Pull request merged**